src/components/Box/Box.jsx

```diff
@@ -0,0 +1,598 @@
+import React, {useRef, forwardRef} from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import useMergeRefs from "hooks/useMergeRefs";
+import {
+    OPACITY,
+    OVERFLOW,
+    OVERFLOW_X,
+    OVERFLOW_Y,
+    BORDER,
+    BORDER_TOP,
+    BORDER_END,
+    BORDER_BOTTOM,
+    BORDER_START,
+    BORDER_COLOR,
+    ROUNDED,
+    ROUNDED_TOP,
+    ROUNDED_END,
+    ROUNDED_BOTTOM,
+    ROUNDED_START,
+    SHADOW,
+    MARGIN,
+    MARGIN_X,
+    MARGIN_Y,
+    MARGIN_TOP,
+    MARGIN_END,
+    MARGIN_BOTTOM,
+    MARGIN_START,
+    MARGIN_NEGATIVE,
+    MARGIN_NEGATIVE_X,
+    MARGIN_NEGATIVE_Y,
+    MARGIN_NEGATIVE_TOP,
+    MARGIN_NEGATIVE_END,
+    MARGIN_NEGATIVE_BOTTOM,
+    MARGIN_NEGATIVE_START,
+    PADDING,
+    PADDING_X,
+    PADDING_Y,
+    PADDING_TOP,
+    PADDING_END,
+    PADDING_BOTTOM,
+    PADDING_START,
+    TEXT_ALIGN,
+    TEXT_TRANSFORM,
+    USER_SELECT,
+    POINTER_EVENTS,
+    VISIBILITY,
+    BACKGROUND_COLORS,
+    COLORS,
+} from "./BoxConstants"
+import styles from "./Box.module.scss";
+
+const Box = forwardRef(
+    ({
+         className,
+         id,
+         children,
+         opacity,
+         overflow,
```

Comment:
for all overflow - remove visible as it is looks like a bug and we don't need to use it